pkg/ui/workspaces/cluster-ui/src/selectors/insightsCommon.selectors.ts

```diff
@@ -26,12 +26,27 @@ export const selectStatementInsightDetailsCombiner = (
   if (!statementInsights || !executionID) {
     return null;
   }
-
   return statementInsights.find(
     insight => insight.statementExecutionID === executionID,
   );
 };
 
+export const selectStatementInsightDetailsCombinerByFingerprint = (
+  statementInsights: StmtInsightEvent[],
+  fingerprintID: string,
+): StmtInsightEvent[] | null => {
+  if (!statementInsights || statementInsights?.length < 1 || !fingerprintID) {
+    return null;
+  }
+  const insightEvents: StmtInsightEvent[] = [];
+  statementInsights.forEach(insight => {
```

Comment:
nit: Try replacing this with an aray.filter, and you can set the var above directly.